test-d/exact.ts

```diff
@@ -143,3 +143,69 @@ import type {Exact} from '../index';
 		fn(input);
 	}
 }
+
```

Comment:
A little bit out of the scope of this PR but can we also include exhaustive tests against ReadonlyArray as well? I just want to ensure ReadonlyArray will work consistently in the future along with the regular mutable arrays.

<img width="886" alt="Screen Shot 2022-07-11 at 9 22 56 am" src="https://user-images.githubusercontent.com/2525002/178165972-37c388ef-9cc6-4eeb-880e-c777e26d5c77.png">

We can just copy & paste Array spec for the ReadonlyArray test